### PR TITLE
microsoft.workloadiam: Refactor subcommand invocation

### PR DESCRIPTION
**Pull request by AntonioND**
The old code was unreliable. The new code should work in a similar way but it should be able to handle errors in a more graceful way.

For example, this is the output in a successful run:
ok

And this is the output in a non-successful run when the user has already used that local authority before and the workload-iam extension fails to create a join token:
not ok

It also adds a check to ensure that the instance name isn't too long. The instance name is used to form labels for Kubernetes resources, and their labels are limited to 63 characters. If the instance name is too long, the labels will go over that limit.

https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/k8s-extension/azext_k8s_extension/partner_extensions/WorkloadIAM.py [hunk 1]
```diff
@@ -44,6 +45,15 @@ def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_t
...
             release_train = 'preview'
 
+        # The name is used as a base to generate Kubernetes labels for config maps, pods, etc, and
+        # their names are limited to 63 characters (RFC-1123). Instead of calculating the exact
+        # number of characters that we can allow users to specify, it's better to restrict that even
+        # more so that we have extra space in case the name of a resource changes and it pushes the
+        # total string length over the limit.
+        if len(name) > 20:
+            raise InvalidArgumentValueError(
+                f"Name '{name}' is too long, it must be 20 characters long max.")
```
Comment (dafang982) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/WorkloadIAM.py:55]: Is it worth declaring a constant and sharing this with the test code?

Comment (AntonioND) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/WorkloadIAM.py:55]: If we do it here, we would have to do it for every string. Also, I'm not sure if f-strings would work in that case, which means that the change could be quite distracting. If we do that, we should do it as part of a different PR to simplify the review of this one.

**Comment by Arif-lakhani** [comment]
Is the 20 characters limit on the name documented anywhere? Is there a TSG that could be appended to the error if so? 
LGTM other than the above comment.

**Review by Arif-lakhani — APPROVED** [review]
LGTM

**Comment by AntonioND** [comment]
> Is the 20 characters limit on the name documented anywhere? Is there a TSG that could be appended to the error if so? LGTM other than the above comment.

We have a TSG that is in the extension metadata ([https://aka.ms/workloadiam-azurearc-tsg](https://aka.ms/workloadiam-azurearc-tsg)), I will add a comment there.

**Review by bavneetsingh16 — APPROVED** [review]

**Pull request merged**